src/libarchive/libarchive.zig

```diff
@@ -616,24 +617,26 @@ pub const Archive = struct {
                                             const read = lib.archive_read_data(archive, plucker_.contents.list.items.ptr, size);
                                             try plucker_.contents.inflate(@as(usize, @intCast(read)));
                                             plucker_.found = read > 0;
-                                            plucker_.fd = bun.toFD(file.handle);
+                                            plucker_.fd = file_handle;
                                             continue :loop;
                                         }
                                     }
                                 }
                                 // archive_read_data_into_fd reads in chunks of 1 MB
                                 // #define	MAX_WRITE	(1024 * 1024)
-                                if (size > 1_000_000) {
-                                    C.preallocate_file(
-                                        file.handle,
-                                        0,
-                                        entry_size,
-                                    ) catch {};
+                                if (comptime Environment.isLinux) {
```

Comment:
do we need the if check. these functions are defined in both darwin_c and windows_c but are empty.

Reply:
Since they aren't doing anything on windows or macos, might as well remove the function call and `if (size < ...)` at comptime